prombench/manifests/cluster-infra/7a_commentmonitor_configmap_noparse.yaml

```diff
@@ -48,3 +48,10 @@ data:
         **Other Commands:**
         To stop benchmark: `/prombench cancel`
         To restart benchmark: `/prombench restart v2.12.0`
+
+    - event_type: funcbench_start
+      regex_string: ^/funcbench (?P<BRANCH>\S+)? (?P<BENCH_FUNC_REGEX>.*)?
```

Comment:
https://golang.org/pkg/testing/#hdr-Benchmarks, `func BenchmarkXxx(*testing.B)` I think it'll be good to add the regex to check it aswell.

Maybe try this regex? 
```
(?mi)^/funcbench\s+(?P<BRANCH>[^\s]+)\s*(?P<BENCH_FUNC_REGEX>Benchmark[^\s]+)?\s*$
```

Reply:
Well, this is tricky. But I guess for 80%, we do all the benchmarks, if you specify it, you absolutely know what is going on there. Plus, if you don't have the

Reply:
> When you specify it, you absolutely know what is going on there. Otherwise, you will see a failed result.

What do you mean by this? Why will one see failed results?

Reply:
This regex will allow:
```
/funcbench my-branch BenchmarkFuncNameHere
```
and
```
/funcbench my-branch
```
and 
```
/funcbench my-branch

some other text in the comment.
```

I was thinking we could use these two ways, the latter one runs all benchmarks? If comment-monitor is not handling empty `BENCH_FUNC_REGEX` as expected this might fail. Will check this and get back; `comment-monitor` is also missing tests. :(

What comment format do you suggest to run all benchmarks?

Reply:
I think these ones you suggested are a very good start

I can't remember exactly, but I think there was a way to run all bench tests for a given package. This would be also useful to add at some point so people can run let's say all bench tests for a single package - promql, tsdb  etc.

Reply:
Sorry, I was responding in a hurry. 
> When you specify it, you absolutely know what is going on there. Otherwise, you will see a failed result.

I mean the `BENCH_FUNC_REGEX` part, even prefix with `Benchmark` in regex, it doesn't stop you making mistakes like `/funcbench master BenchmarkWithTypo.*`. You still can get a failed (or empty ?) benchmark result. PS, we can still `/funcbench branch .*`, right ?

Thanks @geekodour gave some good examples on valid commands to start the funcbench. 😺 
so, my suggestion on the regexp would be
`(?m)^/funcbench[[:blank:]]+(?P<BRANCH>\S+)[[:blank:]]*(?P<BENCH_FUNC_REGEX>\S*)?[[:blank:]]*$`
The testcase to test this regexp is (NOTICE: keep the spaces and newlines)
```
/funcbench master-branch .*

/funcbench evenwith/emoji😅 BenchmarkQuery.*

/funcbench master  BenchmarkQuery.*   
/funcbench your_branch benchmark.*

/funcbench the.dot BenchmarkQuery[Some=\S+  
comments

/funcbench master Benchmark?uery\S+

comments

/funcbench master   
comment comment

// followings are invalid

/funcbench     
comments

/funcbenchmaster BenchmarkQuery.*
comment
```

> This would be also useful to add at some point so people can run let's say all bench tests for a single package - promql, tsdb etc.

I guess this requires to specify the package name instead of `./...`(what we are doing now)

Reply:
So what is the final regex that we're using? I didn't see any change made to the regex? o.O 


> I mean the `BENCH_FUNC_REGEX` part, even prefix with Benchmark in regex, it doesn't stop you making mistakes like `/funcbench master BenchmarkWithTypo.*`. You still can get a failed (or empty ?) benchmark result. PS, we can still `/funcbench branch .*`, right ?

With the regex I suggested you won't be able to do the second one, but the funcbench cli will still be able to do that ofcourse. The idea of having `Benchmark` in the regex itself was to have it fail more to so that it does not run in vain, because once the regex is validated it'll be spinning up GKE nodes. There's always a probability of having a typo in the comment, cannot avoid it but we can always minimize.

`(?mi)^/funcbench\s+(?P<BRANCH>[^\s]+)\s*(?P<BENCH_FUNC_REGEX>Benchmark[^\s]+|\.\*)?\s*$`

^ with this modification, we explicitly allow `.*`

I am not good with regexes so feel free to improve on it, but it works :)